multicluster/cmd/unlink.go

```diff
@@ -70,50 +71,42 @@ func newUnlinkCommand() *cobra.Command {
 			lease := resource.NewNamespaced(coordinationv1.SchemeGroupVersion.String(), "Lease", fmt.Sprintf("service-mirror-write-%s", opts.clusterName), opts.namespace)
 
 			resources := []resource.Kubernetes{
-				secret, link, clusterRole, clusterRoleBinding,
-				role, roleBinding, serviceAccount, serviceMirror, lease,
+				clusterRole, clusterRoleBinding,
+				role, roleBinding, serviceAccount, serviceMirror,
 			}
 
 			if l.Spec.ProbeSpec.Path != "" {
 				gatewayMirror := resource.NewNamespaced(corev1.SchemeGroupVersion.String(), "Service", fmt.Sprintf("probe-gateway-%s", opts.clusterName), opts.namespace)
 				resources = append(resources, gatewayMirror)
 			}
 
-			selector := fmt.Sprintf("%s=%s,%s=%s",
-				k8s.MirroredResourceLabel, "true",
-				k8s.RemoteClusterNameLabel, opts.clusterName,
-			)
-			svcList, err := k.CoreV1().Services(metav1.NamespaceAll).List(cmd.Context(), metav1.ListOptions{LabelSelector: selector})
-			if err != nil {
-				return err
-			}
-			for _, svc := range svcList.Items {
-				resources = append(resources,
-					resource.NewNamespaced(corev1.SchemeGroupVersion.String(), "Service", svc.Name, svc.Namespace),
-				)
-			}
+			if !opts.onlyController {
+				resources = append(resources, secret, link, lease)
 
-			selector = fmt.Sprintf("%s=%s", clusterNameLabel, opts.clusterName)
-			destinationCredentials, err := k.CoreV1().Secrets(controlPlaneNamespace).List(cmd.Context(), metav1.ListOptions{LabelSelector: selector})
-			if err != nil {
-				return err
-			}
-			for _, secret := range destinationCredentials.Items {
-				resources = append(resources,
-					resource.NewNamespaced(corev1.SchemeGroupVersion.String(), "Secret", secret.Name, secret.Namespace),
-				)
+				svcs, err := getServiceMirrors(cmd.Context(), k, opts.clusterName)
+				if err != nil {
+					return err
+				}
+				resources = append(resources, svcs...)
+
+				creds, err := getDestinationCredentials(cmd.Context(), k, opts.clusterName)
+				if err != nil {
+					return err
+				}
+				resources = append(resources, creds...)
 			}
 
 			for _, r := range resources {
-				if opts.output == "yaml" {
+				switch opts.output {
+				case "yaml":
 					if err := r.RenderResource(stdout); err != nil {
 						log.Errorf("failed to render resource %s: %s", r.Name, err)
 					}
-				} else if opts.output == "json" {
+				case "json":
 					if err := r.RenderResourceJSON(stdout); err != nil {
 						log.Errorf("failed to render resource %s: %s", r.Name, err)
 					}
-				} else {
+				default:
```

Comment:
obviously this isn't a bad change on its own, but its unrelated to this functionality change. would be better to omit.